### ISSUE TITLE
Firefox opened with keyboard shortcut does not get focus

### ISSUE THREAD

**Issue by photong** [linuxmint/mint21-beta#160]
When Firefox is launched with a keyboard shortcut, the focus remains on the window that had focus before Firefox was launched.

Steps to reproduce:

1. Launch terminal.
2. Open Firefox with keyboard shortcut (for me it's super + b).
3. Note that Firefox window does not have focus (terminal has focus).

Expected behavior:
I expect a newly opened window to have focus.

This frustrating bug was present in Mint 20.

**Comment by clefebvre** [comment]
I think that's by design specifically because it's run from terminal. @mtwebster can you confirm?

**Comment by clefebvre** [comment]
I can't reproduce this. Newly mapped Firefox takes focus. What doesn't is a new tab as a result of Firefox already having an instance.

**Comment by LinuxOnTheDesktop** [comment]
Cf. [this report](https://github.com/linuxmint/cinnamon/issues/10387) filed against (in effect) Mint 20. Is the problem really solved? I tried just now with Mint 21, on a live USB (from several days ago) and the problem seemed to be present. Perhaps the previously-focused window was no longer focused (that previous window was a terminal, and Firefox was maximised when it [edit: by 'it' I mean Firefox] appeared) but Firefox's URL bar was not focused.

**Comment by photong** [comment]
Attached is a recording of the problem. I launch Nemo using keyboard shortcut (note how Nemo has focus). I then launch Firefox using keyboard shortcut (note how Firefox appears behind Nemo). Launching Firefox using the panel icon works as expected.

![out](https://user-images.githubusercontent.com/57975239/181052439-79749589-4bd0-45fa-a553-91aa333591a4.gif)

**Comment by mtwebster** [comment]
I can reproduce, but it doesn't happen always, we're trying to figure out what the criteria are. It doesn't always happen -

For instance,
- It never occurs when there aren't any firefox windows already open.
- It doesn't always occur with multiple monitors, depending on where the pointer or focus currently is.
- From a bit of googling, this isn't unique to Cinnamon, though it definitely does not happen everywhere.

No one has said it's solved, and this bug is still open.

**Comment by ghost** [comment]
I can't reproduce this on a fresh install of LM 20.0 - Cinnamon 4.6.7. However, reports ensure that it is present from LM 20.1 to LM 21.0

**Comment by photong** [comment]
I can reproduce this on the beta image.

1. Boot beta image.
2. Go to "Keyboard" -> "Shortcuts".
3. Click on "Launchers" and set new shortcut for "Launch web browser" (e.g., super + b)
4. Close Keyboard window.
5. Press shortcut for web browser (super + b); on first launch, Firefox has focus.
6. Close Firefox and press shortcut for web browser again; this time, window does not have focus (the "x" on top right is not green).
7. With Firefox still NOT in focus, press shortcut; another window opens without focus.

Pressing super + b will keep spawning new windows, none with focus.

The interesting part is that if you click on a Firefox window and press the shortcut, the new Firefox window WILL have focus (this does not work on installed beta with updates though).

To summarize, on beta image:

- If Firefox window exists and has focus, super + b creates new Firefox window with focus.
- If Firefox window exists and does not have focus, super + b creates new Firefox window without focus.
- If any other window (e.g., Nemo) exists and has focus, super + b creates new Firefox window without focus (focus never leaves Nemo).

Launching from panel (with ctrl + click) always works as expected.
